hnn_core/tests/test_network.py

```diff
@@ -91,36 +88,41 @@ def test_network():
             assert set(conn['gid_pairs'].keys()) == set(
                 net.external_drives[src_type]['conn'][target_type]['src_gids'])
 
-    assert len(net.gid_ranges['bursty1']) == 1
+    # Check drive dict structure for each external drive
     for drive in net.external_drives.values():
+        # Check that connectivity sources correspond to gid_ranges
+        this_src_gids = set([gid for drive_conn in
+                             drive['conn'].values() for
+                             gid in drive_conn['src_gids']])  # NB set: globals
+        assert sorted(this_src_gids) == list(net.gid_ranges[drive['name']])
+        # Check type-specific dynamics and events
+        n_drive_cells = drive['n_drive_cells']
         assert len(drive['events']) == 1  # single trial simulated
         if drive['type'] == 'evoked':
             for kw in ['mu', 'sigma', 'numspikes']:
                 assert kw in drive['dynamics'].keys()
-            assert len(drive['events'][0]) == net.n_cells
+            assert len(drive['events'][0]) == n_drive_cells
             # this also implicitly tests that events are always a list
             assert len(drive['events'][0][0]) == drive['dynamics']['numspikes']
         elif drive['type'] == 'gaussian':
             for kw in ['mu', 'sigma', 'numspikes']:
                 assert kw in drive['dynamics'].keys()
-            assert len(drive['events'][0]) == net.n_cells
+            assert len(drive['events'][0]) == n_drive_cells
         elif drive['type'] == 'poisson':
             for kw in ['tstart', 'tstop', 'rate_constant']:
                 assert kw in drive['dynamics'].keys()
-            assert len(drive['events'][0]) == net.n_cells
-
+            assert len(drive['events'][0]) == n_drive_cells
         elif drive['type'] == 'bursty':
             for kw in ['tstart', 'tstart_std', 'tstop',
-                       'burst_rate', 'burst_std', 'numspikes', 'repeats']:
+                       'burst_rate', 'burst_std', 'numspikes']:
                 assert kw in drive['dynamics'].keys()
-            assert len(drive['events'][0]) == 1
+            assert len(drive['events'][0]) == n_drive_cells == 10
```

Comment:
```suggestion
            assert len(drive['events'][0]) == n_drive_cells == repeats
```

? so it's clear for a future contributor

Reply:
I guess what I meant here was that `n_drive_cells` should be defined before the function call somewhere? But maybe it's not possible because the value is implicitly extracted from param file? I would still do something like:

```py
n_drive_cells = 10

net.external_drives['drive_name']['n_drive_cells'] = n_drive_cells

assert len(drive['events'][0]) == n_drive_cells
```

or something of this sort

Reply:
I don't see where the `repeats` parameter specified? Or are you proposing that we create this variable anew?

Reply:
Oh sorry, I just refreshed my page and saw your most recent post. As you suspected, `n_drive_cells` is extracted from the param file here. I did add a test to test_drives.py that tests whether the `n_drive_cells` value set by the user is reflected in `net.external_drives['drive_name']['n_drive_cells']`. Does that fit the bill?

Reply:
I guess I'm just trying to avoid comparison to a hardcoded number in the test. Maybe this is enough:

```suggestion
            assert len(drive['events'][0]) == n_drive_cells
```